app/modules/kit-library/module.php

```diff
@@ -75,6 +78,21 @@ private function set_kit_library_settings() {
 		] );
 	}
 
+	private function apply_filter_subscription_plans( array $subscription_plans ): array {
+		foreach ( $subscription_plans as $key => $plan ) {
+			if ( $plan[ 'promotion_url' ] == null) {
+				continue;
+			}
+			$subscription_plans[ $key ] = Filtered_Promotions_Manager::get_filtered_promotion_data(
```

Comment:
```suggestion
			if ( null === $plan[ 'promotion_url' ] ) {
				continue;
			}

			$subscription_plans[ $key ] = Filtered_Promotions_Manager::get_filtered_promotion_data(
```
Or wrap everything in `if( isset( $plan[ 'promotion_url' ] ) ) { ... }`